### PR TITLE
Fix inheritence for strokes and fills, particularly the w3 svg logo test

### PR DESCRIPTION
**Pull request by dnfield**
Does the following:

- Avoid trying to cascade inherited attributes down the tree at parse time. Instead, do it at resolve time.
- Handle defaulting filling for shape/text elements until resolve time, when we know if this shape needs a color applied from a parent or a `use` element.
- Adds a test covering the `use` case that previously didn't work.

This addresses the second SVG in https://github.com/dnfield/flutter_svg/issues/864, but not the first. The first I believe is a different bug related to stroking width and pre-calculating transforms.

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/vector_graphics_compiler/lib/src/svg/node.dart [hunk 1]
```diff
@@ -426,11 +449,11 @@ class TextNode extends AttributedNode {
 
   /// Compute the [Paint] that this text node uses.
   Paint? computePaint(Rect bounds, AffineMatrix transform) {
-    final Fill? fill = attributes.fill?.toFill(bounds, transform);
+    final Fill? fill = attributes.fill?.toFill(bounds, transform, defaultColor: Color.opaqueBlack);
     final Stroke? stroke = attributes.stroke?.toStroke(bounds, transform);
-    if (fill == null && stroke == null) {
-      return null;
-    }
+    // if (fill == null && stroke == null) {
```
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/vector_graphics_compiler/lib/src/svg/node.dart:454]: This needs a test. Argh.

File: packages/vector_graphics_compiler/lib/src/svg/parser.dart [hunk 2]
```diff
@@ -1492,9 +1489,9 @@ class SvgParser {
 
   Map<String, String> _createAttributeMap(List<XmlEventAttribute> attributes) {
     final Map<String, String> attributeMap = <String, String>{};
-    if (_parentDrawables.isNotEmpty && currentGroup != null) {
-      attributeMap.addEntries(currentGroup!.attributes.heritable);
-    }
+    // if (_parentDrawables.isNotEmpty && currentGroup != null) {
```
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/vector_graphics_compiler/lib/src/svg/parser.dart:1492]: This needs deletion.

File: packages/vector_graphics_compiler/lib/src/svg/node.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/vector_graphics_compiler/lib/src/svg/node.dart]: Added a test covering this so the code wouldn't accidentally get deleted again.

**Comment by dnfield** [comment]
Linux is giving slightly different values for the cubic calculations.

**Comment by dnfield** [comment]
This is broken for strokes that have opacity values.

**Pull request merged**